src/components/TagSelectorComp.tsx

```diff
@@ -44,163 +48,180 @@ export interface Option {
+        };
     }
-    if (
-      actionMeta.action === 'create-option'
-    ) {
-      setIsLoading(true);
-      inputValue[inputValue.length - 1].label = inputValue[
-        inputValue.length - 1
-      ].label.toLowerCase();
-      try {
-        if(props.tagLabel.status==='available'){
-          props.tagLabel.setValue(inputValue[inputValue.length - 1].label);
-        }
-        if(props.createTag.canExecute){
-          props.createTag.execute();
-        }
-        setLabels(inputValue);
-      } catch (err) {
-        console.error('Failed to create a Tag: ' + err);
-      }
-      setIsLoading(false);
+    if (props.enableCreate) {
+        return (
+            <CreatableSelect
+                isMulti
+                options={options}
+                value={labels}
+                onChange={handleChange}
+                isLoading={isLoading}
+                components={props.animatedDelete ? animatedComponents : undefined}
+                styles={styles}
+                placeholder={props.placeholder}
+                className={props.className!}
+                classNamePrefix={props.classNamePrefix}
+                tabSelectsValue={false}
+            />
+        );
     }
-    if (
-      actionMeta.action === 'remove-value' ||
-      actionMeta.action === 'pop-value'
-    ) {
-      setIsLoading(true);
-      try {
-        if(props.tagLabel.status==='available'){
-          props.tagLabel.setValue(actionMeta.removedValue.label);
+    else {
+        return (
+            <Select
+                isMulti
+                options={options}
+                value={labels}
+                onChange={handleChange}
+                isLoading={isLoading}
+                components={props.animatedDelete ? animatedComponents : undefined}
+                styles={styles}
+                placeholder={props.placeholder}
+                className={props.className!}
+                classNamePrefix={props.classNamePrefix}
+                tabSelectsValue={false}
+            />
+        );
+    }
+
+
+    function clearAction() {
+        setIsLoading(true);
+        try {
+            if (props.removeAllTags.canExecute) {
+                props.removeAllTags.execute();
+            }
+            setLabels([]);
+        } catch (err) {
+            console.error('Failed to remove all Tags: ' + err);
         }
-        if(props.removeTag.canExecute){
-          props.removeTag.execute();
+        setIsLoading(false);
+    }
+
+    function removeAction(actionMeta: any, inputValue: any) {
+        setIsLoading(true);
+        try {
+            if (props.tagLabel.status === 'available') {
+                props.tagLabel.setValue(actionMeta.removedValue.label);
+            }
+            if (props.removeTag.canExecute) {
+                props.removeTag.execute();
+            }
+            setLabels(inputValue);
+        } catch (err) {
+            console.error('Failed to remove a Tag: ' + err);
```

Comment:
Would it cause some exception in the app itself if widget is throwing the exception? Normally exception will appear in the console anyway. I'd remove all catches, it will confuse a user. Something should tell him that funtionality isn't working, just silently swallowing the error isn't good.

Reply:
done